video_creation/final_video.py

```diff
@@ -4,37 +4,44 @@
 import re
 from os.path import exists
 from typing import Tuple, Any
-from moviepy.audio.AudioClip import concatenate_audioclips, CompositeAudioClip
-from moviepy.audio.io.AudioFileClip import AudioFileClip
-from moviepy.video.VideoClip import ImageClip
-from moviepy.video.compositing.CompositeVideoClip import CompositeVideoClip
-from moviepy.video.compositing.concatenate import concatenate_videoclips
-from moviepy.video.io.VideoFileClip import VideoFileClip
+
+from moviepy.editor import (
+    VideoFileClip,
+    AudioFileClip,
+    ImageClip,
+    concatenate_videoclips,
+    CompositeAudioClip,
+    CompositeVideoClip,
+)
 from moviepy.video.io.ffmpeg_tools import ffmpeg_extract_subclip
 from rich.console import Console
+from rich.progress import track
 
 from utils.cleanup import cleanup
 from utils.console import print_step, print_substep
 from utils.videos import save_data
 from utils import settings
+from video_creation.background import download_background, chop_background_video
 
 console = Console()
-W, H = 1080, 1920
 
 
-def name_normalize(name: str) -> str:
+def name_normalize(
+        name: str
+) -> str:
     name = re.sub(r'[?\\"%*:|<>]', "", name)
-    name = re.sub(r"( [w,W]\s?\/\s?[o,O,0])", r" without", name)
-    name = re.sub(r"( [w,W]\s?\/)", r" with", name)
-    name = re.sub(r"(\d+)\s?\/\s?(\d+)", r"\1 of \2", name)
-    name = re.sub(r"(\w+)\s?\/\s?(\w+)", r"\1 or \2", name)
-    name = re.sub(r"\/", r"", name)
-    name[:30]
-
-    lang = settings.config["reddit"]["thread"]["post_lang"]
+    name = re.sub(r'( [w,W]\s?\/\s?[o,O,0])', r' without', name)
+    name = re.sub(r'( [w,W]\s?\/)', r' with', name)
+    name = re.sub(r'(\d+)\s?\/\s?(\d+)', r'\1 of \2', name)
+    name = re.sub(r'(\w+)\s?\/\s?(\w+)', r'\1 or \2', name)
+    name = re.sub(r'\/', '', name)
+    name[:30]  # the hell this little guy does?
```

Comment:
It fixes both an FFMPEG file error and some OS filing errors

Reply:
it do nothing

can you show me an example? maybe I'm just blind or something

Reply:
@JasonLovesDoggo 
if you want to make a substring with first 30 chars you need to redefine it
```python
name = name[:30]
```